src/test/java/it/smartcommunitylab/aac/saml/SamlIdentityProviderConfigurationTest.java

```diff
@@ -0,0 +1,320 @@
+            idpCfgMap.setConfiguration(rawCfgMap);
+
+            Assertions.assertThat(idpCfgMap.getEntityId()).isEqualTo(rawCfgMap.get("entityId"));
+            Assertions.assertThat(idpCfgMap.getSigningKey()).isEqualTo(rawCfgMap.get("signingKey"));
+            Assertions.assertThat(idpCfgMap.getSigningCertificate()).isEqualTo(rawCfgMap.get("signingCertificate"));
+            Assertions.assertThat(idpCfgMap.getCryptKey()).isEqualTo(rawCfgMap.get("cryptKey"));
+            Assertions.assertThat(idpCfgMap.getCryptCertificate()).isEqualTo(rawCfgMap.get("cryptCertificate"));
+
+            Assertions.assertThat(idpCfgMap.getIdpMetadataUrl()).isEqualTo(rawCfgMap.get("idpMetadataUrl"));
+            Assertions.assertThat(idpCfgMap.getIdpEntityId()).isEqualTo(rawCfgMap.get("idpEntityId"));
+            Assertions.assertThat(idpCfgMap.getWebSsoUrl()).isEqualTo(rawCfgMap.get("webSsoUrl"));
+            Assertions.assertThat(idpCfgMap.getWebLogoutUrl()).isEqualTo(rawCfgMap.get("webLogoutUrl"));
+            Assertions.assertThat(idpCfgMap.getSignAuthNRequest()).isEqualTo(rawCfgMap.get("signAuthNRequest"));
+            Assertions.assertThat(idpCfgMap.getVerificationCertificate()).isEqualTo(rawCfgMap.get("verificationCertificate"));
+            Assertions.assertThat(idpCfgMap.getSsoServiceBinding()).isEqualTo(rawCfgMap.get("ssoServiceBinding"));
+
+            Assertions.assertThat(idpCfgMap.getNameIDFormat()).isEqualTo(rawCfgMap.get("nameIDFormat"));
+            Assertions.assertThat(idpCfgMap.getNameIDAllowCreate()).isEqualTo(rawCfgMap.get("nameIDAllowCreate"));
+            Assertions.assertThat(idpCfgMap.getForceAuthn()).isEqualTo(rawCfgMap.get("forceAuthn"));
+            Assertions.assertThat(idpCfgMap.getIsPassive()).isEqualTo(rawCfgMap.get("isPassive"));
+            Assertions.assertThat(idpCfgMap.getAuthnContextClasses()).isEqualTo(rawCfgMap.get("authnContextClasses"));
+            Assertions.assertThat(idpCfgMap.getAuthnContextComparison()).isEqualTo(rawCfgMap.get("authnContextComparison"));
+            Assertions.assertThat(idpCfgMap.getUserNameAttributeName()).isEqualTo(rawCfgMap.get("userNameAttributeName"));
+
+            Assertions.assertThat(idpCfgMap.getTrustEmailAddress()).isEqualTo(rawCfgMap.get("trustEmailAddress"));
+            Assertions.assertThat(idpCfgMap.getAlwaysTrustEmailAddress()).isEqualTo(rawCfgMap.get("alwaysTrustEmailAddress"));
+            Assertions.assertThat(idpCfgMap.getRequireEmailAddress()).isEqualTo(rawCfgMap.get("requireEmailAddress"));
+        }
+    }
+
+    /**
+     * This tests checks if s saml configuration map matches the content of the given saml configuration
+     * @throws Exception
+     */
+    @Test
+    public void validateSamlIdentityProviderConfigCreation() throws Exception {
+
+        // Create (hardcoded) configuration map to be used for initialization
+        Map<String, Serializable> rawCfgMap = buildRawConfigurationMap(true);
```

Comment:
given that this test **checks** the config and the map do not use the helper function, but build the map here and then check